mozilla_schema_generator/common_ping.py

```diff
@@ -50,32 +50,37 @@ def with_description(dtype: dict, comment: str) -> dict:
         active_addons = prepend_properties(("environment", "addons", "activeAddons")) \
             + ("additionalProperties", "properties")
 
-        schema.set_schema_elem(
-            prepend_properties(("environment", "settings", "userPrefs")),
-            with_description(
-                string_map,
-                "User preferences - limited to an allowlist defined in `toolkit/components/telemetry/app/TelemetryEnvironment.jsm`",  # NOQA
-            ),
-        )
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "version")), string)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "hasSuperfetch")), boolean)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "hasPrefetch")), boolean)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "addons", "theme", "foreignInstall")), integer)
-        schema.set_schema_elem(active_addons + ("foreignInstall",), integer)
-        schema.set_schema_elem(active_addons + ("version",), string)
-        schema.set_schema_elem(active_addons + ("userDisabled",), integer)
-        schema.set_schema_elem(
-            prepend_properties(("environment", "addons", "activeGMPlugins"))
-            + ("additionalProperties", "properties", "applyBackgroundUpdates"),
-            with_description(
-                integer,
-                "Cast into an integer via mozilla-schema-generator. See bug 1611027.",
-            ),
-        )
+        # If active_addons doesn't exist in the schema, then this is a simplified
+        # environment and we can skip the below modifications.
+        if property_exists(schema.schema, active_addons):
```

Comment:
Pipeline metadata is new enough that I'm not quite ready to think about how to annotate the environment field using that mechanism. But it would make sense to be a bit pickier here, and verify existence of specific sections of the environment before modifying them, rather than using active_addons as an indicator of a full environment.

Things get messy with the "os" modifications, though. I don't believe those are necessary and I'm pursing replacing them with fields in the source json schemas in https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/600